Data/ByteString.hs

```diff
@@ -1405,11 +1405,14 @@ breakSubstring pat =
 findSubstring :: ByteString -- ^ String to search for.
               -> ByteString -- ^ String to seach in.
               -> Maybe Int
-findSubstring pat src
-    | null pat && null src = Just 0
-    | null b = Nothing
-    | otherwise = Just (length a)
-  where (a, b) = breakSubstring pat src
+findSubstring pat = go
+  where
+    breakPat = breakSubstring pat
+    go src
+      | null pat && null src = Just 0
+      | null b = Nothing
+      | otherwise = Just (length a)
+      where (a, b) = breakPat src
```

Comment:
I'm not desperately fond of this `f x = go` pattern when the `go` body is not recursive, it's just the normal body.

I think I'd prefer `let` and lambda.

The more important thing here is to comment that the `breakPat = breakSubstring pat` bit is there to memoise the work that can be done on the pattern independent of the src.

Reply:
I can use `let` and lambda, but without `MultiWayIf`, I can't see how to avoid ending up with another name somewhere. Should I add `MultiWayIf` to the language pragmas?